### PR TITLE
Create yml file 

### PR DESCRIPTION
**Pull request by juan-castrillon**
### Created the YML file used to trigger and control Github actions.

How the different commands work is explained in the comments of the file, 

However, after running some test in a repository of my own, with a simple app the test took ~40s. Its important to remember that [we only have 2000 minutes of test a month](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions).

In order to make the execution time faster, I added the cache action which is supposed to help. If we think that we need it to be even faster, that could be a topic for future work . (Evaluate the JDK setting, and the gradle running, which takes all of the time)

### CODE REVIEW

**Comment by juan-castrillon** [comment]
Also, as soon as this gets merged, I will submit the request that closes #11 , and you guys will be able to see the test being performed on the new code

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: 
+      - master
+    paths:
+      - 'src/**'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (alyhasansakr) [inline .github/workflows/gradle.yml:14]:
```suggestion
    runs-on: windows-latest
```
We are mainly using windows now, later we are going to add Linux as well

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2 ##Copies the repository files in the remote machine
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1 #Sets up the Java version (8)
+      with:
+        java-version: 1.8
```
Comment (alyhasansakr) [inline .github/workflows/gradle.yml:21]: any specific reason we should use this version of java?

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (juan-castrillon) [reply .github/workflows/gradle.yml:14]: Yes, but this is only the GitHub hosted virtual machine in which the action gets executed. (I mean it doesn't influence our development at all) so  I thought of going with the default configuration. Do we really need to differentiate here?

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (juan-castrillon) [reply .github/workflows/gradle.yml:21]: Not really. I just went with default again. Is there any specific version of Java we should define? I tought we could keep this one, and then modify it if we need any functionality of a higher version. But do you think we should configure it to any specific one?

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (alyhasansakr) [reply .github/workflows/gradle.yml:14]: normally not, but if we do something platform dependent (like network sockets), then we will need to differentiate. Anyway we can keep it as it is now and see if it causes problems in the future.

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (alyhasansakr) [reply .github/workflows/gradle.yml:21]: ok, lets just default for now.

**Pull request merged**